test/unit/processing/components/get-remaining-amount.test.js

```diff
@@ -1,19 +1,103 @@
+const { convertToPounds } = require('../../../../app/utility')
+
 const getRemainingAmount = require('../../../../app/processing/components/get-remaining-amount')
-const scheduleDates = JSON.parse(JSON.stringify(require('../../../mock-objects/mock-schedule-periods')))
+
+const scheduleDates = JSON.parse(JSON.stringify(require('../../../mock-objects/mock-payment-timelines')))
+
+let currentClaimValue
+let paidAmount
+let originalRemainingClaimValue
+
+let paymentSchedule
+let newValue
 
 describe('get remaining amount', () => {
-  test('should return remaining amount as number', () => {
-    const result = getRemainingAmount(scheduleDates)
-    expect(result).toEqual(expect.any(Number))
+  beforeEach(() => {
+    currentClaimValue = scheduleDates.reduce((runningTotal, record) => runningTotal + record.value, 0)
+    paidAmount = scheduleDates.filter(record => !record.outstanding).reduce((runningTotal, record) => runningTotal + record.value, 0)
+    originalRemainingClaimValue = currentClaimValue - paidAmount
   })
 
-  test('should return sum of the property value of all items in scheduleDates', () => {
-    const result = getRemainingAmount(scheduleDates)
-    expect(result).toBe(750)
+  describe('top up', () => {
+    beforeEach(() => {
+      paymentSchedule = scheduleDates
+      newValue = currentClaimValue + 25000
+    })
+
+    test('should return a number', () => {
+      const result = getRemainingAmount(paymentSchedule, newValue)
+      expect(result).toEqual(expect.any(Number))
+    })
+
+    test('should return a value greater than Number(convertToPounds(originalRemainingClaimValue))', () => {
+      const result = getRemainingAmount(paymentSchedule, newValue)
+      expect(result).toBeGreaterThan(Number(convertToPounds(originalRemainingClaimValue)))
+    })
+
+    test('should return Number(convertToPounds(newValue - paidAmount))', () => {
+      const result = getRemainingAmount(paymentSchedule, newValue)
+      expect(result).toBe(Number(convertToPounds(newValue - paidAmount)))
+    })
   })
 
-  test('returns zero (0) if no schedule date', () => {
-    const result = getRemainingAmount([])
-    expect(result).toBe(0)
+  describe('reduction', () => {
+    beforeEach(() => {
+      paymentSchedule = scheduleDates
```

Comment:
From prev comment, this and those below need changed to use the corresponding `scheduleDates` for reduction, 0 reduction and recovery